cmd/pbackend/statuser.go

```diff
@@ -153,16 +153,24 @@ func checkSourceAvailabilityAWS(ctx context.Context) {
 				Identity:     s.Identity,
 				ResourceType: "Application",
 			}
-			_, err = clients.GetEC2Client(ctx, &s.Authentication, "")
+			ec2Client, err := clients.GetEC2Client(ctx, &s.Authentication, "")
 			if err != nil {
 				sr.Status = kafka.StatusUnavailable
 				sr.Err = err
 				logger.Warn().Err(err).Msg("Could not get aws assumed client")
-				chSend <- sr
 			} else {
-				sr.Status = kafka.StatusAvaliable
-				chSend <- sr
+				sr.Status = kafka.StatusAvailable
 			}
+			permissions, err := ec2Client.CheckPermission(ctx, &s.Authentication)
+			if err != nil {
+				sr.Err = err
+				logger.Warn().Err(err).Msg("AWS permission check failed")
+			}
+			sr.MissingPermissions = permissions
+			if permissions != nil {
+				logger.Warn().Bool("missing_permission", true).Msgf("No sufficient permissions: %s", permissions)
```

Comment:
Perhaps "insufficient" is better wording.

Also you can use `strings.Join` to join string slice by comma, but [this works too](https://go.dev/play/p/yVujHTNW-IL).

Finally, when `CheckPermission` returns `nil` it likely returns an error too, can you squash this into just a single warning message? You can stuff everything into one log record, me as someone who is debugging this issue would be interested in:

* source id (already present in the context logger)
* account number (could be added in the `processMessage` too as it is in the identity but perhaps out of scope of this PR)
* error message (via Err function)
* missing permissions

So basically this line contains it all, perhaps just delete the 167 and move this into the if statement above or something like that.

Reply:
This is a very good point. The CheckPermission function returned only missing permissions when something was missing, and the error value was returned only when something else failed. We should return an error for missing permissions, so I changed it in this PR. I hope this now makes more sense.